### PR TITLE
Release v1.4.1

### PR DESCRIPTION
**Pull request by balajisankar15**
*Issue #, if available:* https://github.com/aws/sagemaker-distribution/milestone/10

*Description of changes:* v1.4.1 release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by balajisankar15** [comment]
Running tests right now. Will add the staleness report afterwards.

**Comment by balajisankar15** [comment]
python ./src/main.py generate-staleness-report --target-patch-version $TARGET_VERSION

# Staleness Report: 1.4.1(gpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.3|1.26.3
jinja2|3.1.3|3.1.3
pandas|2.1.4|2.1.4
altair|5.2.0|5.2.0
${\color{red}boto3}$|1.28.64|1.28.85
ipython|8.20.0|8.20.0
jupyter-lsp|2.2.2|2.2.2
jupyterlab|4.0.11|4.0.11
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.4.0|2.4.0
amazon-sagemaker-jupyter-scheduler|3.0.5|3.0.5
scipy|1.11.4|1.11.4
scikit-learn|1.3.2|1.3.2
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.2|0.8.2
ipywidgets|8.1.1|8.1.1
notebook|7.0.7|7.0.7
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
fastapi|0.103.2|0.103.2
${\color{red}langchain}$|0.0.350|0.0.354
jupyter-ai|2.8.1|2.8.1
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.0|4.1.0
jupyterlab-git|0.50.0|0.50.0
jupyterlab-lsp|5.0.2|5.0.2
keras|2.12.0|2.12.0
matplotlib|3.8.2|3.8.2
nodejs|18.18.2|18.18.2
py-xgboost-gpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.9.0|1.9.0
pytorch-gpu|2.0.0|2.0.0
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.198.0|2.198.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow|2.12.1|2.12.1
uvicorn|0.25.0|0.25.0

# Staleness Report: 1.4.1(cpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.3|1.26.3
jinja2|3.1.3|3.1.3
pytorch|2.0.0|2.0.0
pandas|2.1.4|2.1.4
altair|5.2.0|5.2.0
${\color{red}boto3}$|1.28.64|1.28.85
ipython|8.20.0|8.20.0
jupyter-lsp|2.2.2|2.2.2
jupyterlab|4.0.11|4.0.11
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.4.0|2.4.0
amazon-sagemaker-jupyter-scheduler|3.0.5|3.0.5
scipy|1.11.4|1.11.4
scikit-learn|1.3.2|1.3.2
pip|23.3.2|23.3.2
torchvision|0.15.2|0.15.2
autogluon|0.8.2|0.8.2
ipywidgets|8.1.1|8.1.1
notebook|7.0.7|7.0.7
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
fastapi|0.103.2|0.103.2
${\color{red}langchain}$|0.0.350|0.0.354
jupyter-ai|2.8.1|2.8.1
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.0|4.1.0
jupyterlab-git|0.50.0|0.50.0
jupyterlab-lsp|5.0.2|5.0.2
keras|2.12.0|2.12.0
matplotlib|3.8.2|3.8.2
nodejs|18.18.2|18.18.2
py-xgboost-cpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.9.0|1.9.0
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.198.0|2.198.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
tensorflow|2.12.1|2.12.1
uvicorn|0.25.0|0.25.0

**Review by claytonparnell — APPROVED** [review]

**Review by trajanikant — APPROVED** [review]
